api/src/main/java/io/opencensus/trace/export/SpanData.java

```diff
@@ -70,7 +71,7 @@ public static SpanData create(
       Timestamp startTimestamp,
       Attributes attributes,
       TimedEvents<Annotation> annotations,
-      TimedEvents<NetworkEvent> networkEvents,
+      TimedEvents<MessageEvent> messageEvents,
```

Comment:
Thanks @sebright. I initially thought this is only used by our impl_core but maybe I am wrong.

This is the most tricky part in this issue. My thoughts here are:

* Add another create() with `TimeEvents<MessageEvent>` as input in parallel: CANNOT be done due to Java compiler's type erasure. The added method will be considered to have the same signature.
* Make a superclass of `MessageEvent` and `NetworkEvent`: I initially thought about that but got trapped by the enum of `NetworkEvent.Type` (enum can not be inherited). So if we do so, we still have to break the compatibility of using NetworkEvent.Type (especially that MessageEvent.Type changed the name of "RECV" to "RECEIVED".
* Make one message extend another: same issue with above - users will end up using MessageEvent with NetworkEvent.Type (or the other way around).

The enum issue can somehow be worked around by changing it to a constant. Though it is not a recommended way.

Reply:
I thought that the common superclass could have a package-private constructor and no methods.  Then SpanData.create could assume that the subclass is either MessageEvent or NetworkEvent (because no one else would be able to extend Event) and use instanceof/casting to handle the two cases.  Then the two enums could remain independent.

Reply:
I can keep `SpanData.create` signature unchanged if I address your next comment (use `NetworkEvent` in implementation). So this change should not be a problem.

Reply:
I will not change it. The old API should be preserved. Instead adding another method with different name and signature (e.g. `createWithMessageEvents`) to get around the type erase check during compilation.